interlinq/utils/vqe_subroutines.py

```diff
@@ -11,23 +13,24 @@ def string_to_qutip_pauli(string):
     if string == 'PauliZ':
         return sigmaz()
 
-def expectation_value(terms, matrices, tmp):
+def expectation_value(terms, vector, number_of_qubits):
     total = 0
-    
+
     for term in terms:
         coefficient, observables = term
-        
-        needed_matrices = []
-        needed_paulis = []
+
+        obs_tensor_prod = []
+
+        for _ in range(number_of_qubits):
+            obs_tensor_prod.append(identity(2))
         
         for obs in observables:
             pauli, index = obs
             
-            needed_matrices.append(matrices[index])
-            needed_paulis.append(string_to_qutip_pauli(pauli))
+            obs_tensor_prod[index] = string_to_qutip_pauli(pauli)
+
+        obs_tensor_prod = tensor(obs_tensor_prod)
         
-        expectation = expect(needed_paulis, needed_matrices)
+        total += coefficient * np.vdot(vector, obs_tensor_prod @ vector) 
         
-        total += coefficient * expectation[0][0]
-    
     return total
```

Comment:
A new line at the end of the file is missing.

Reply:
done